app/models/assignment.rb

```diff
@@ -600,6 +600,20 @@ def summary_json(user)
       numMarked: self.get_num_marked(user.admin? ? nil : user.id) }
   end
 
+  # Generate a JSON summary of the test results associated with an assignment
+  def summary_test_result_json(user)
+    return {} unless user.admin? || user.ta?
+    {
+      submission_test_results: self.submissions.map do |submission|
```

Comment:
Use `self.current_submissions_used` instead of `self.submissions`. A group can have more than one collected submission, but generally we only want to most recent one, which is what `current_submissions_used` gives us.